### PR TITLE
Save power by adding a mandatory sleep between tdlib requests

### PR DESCRIPTION
**Pull request by mbarashkov**
0.02 second in foreground more and 0.1 second in background mode.
This also considerably improves UI smoothness in my tests.

### CODE REVIEW

File: src/tdlibreceiver.cpp [hunk 1]
```diff
@@ -205,6 +214,7 @@ void TDLibReceiver::receiverLoop()
           VERBOSE("Raw result:" << receivedJsonDocument.toJson(QJsonDocument::Indented).constData());
           processReceivedDocument(receivedJsonDocument);
       }
+      sleep(this->powerSavingMode ? POWERSAVING_TDLIB_REQUEST_INTERVAL : NORMAL_TDLIB_REQUEST_INTERVAL);
```
Comment (monich) [inline src/tdlibreceiver.cpp]: What is this sleep that takes `double` as an argument? I suggest to use [msleep](https://doc.qt.io/qt-5/qthread.html#msleep) instead, at least it's clear what it's doing and which units it takes.

File: qml/harbour-fernschreiber.qml [hunk 2]
```diff
@@ -30,6 +30,14 @@ ApplicationWindow
...
     allowedOrientations: defaultAllowedOrientations
 
+    onApplicationActiveChanged: {
+        if(applicationActive) {
+            tdLibWrapper.handleForeground()
+        } else {
+            tdLibWrapper.handleBackground()
+        }
+    }
+
```
Comment (monich) [inline qml/harbour-fernschreiber.qml]: There's no need to involve QML in this, the native `TDLibWrapper` object can directly receive `QGuiApplication::applicationStateChanged` signals from `qGuiApp`.

**Comment by mbarashkov** [comment]
Will address @monich

**Comment by mbarashkov** [comment]
@monich feedback addressed

**Comment by mbarashkov** [comment]
Also my tests showed that throttling requests in foreground is bad because it affects app performance significantly. So we should do it in background.

**Review by monich — APPROVED** [review]
LGTM

**Comment by monich** [comment]
Oh, and one more thing, Could you please squash the commits and format the commit message so that it looks nice in the git history.

**Comment by mbarashkov** [comment]
@monich while fighting with the Github Desktop app etc I've completely messed up the branch. I will make a new pull request.

**Comment by mbarashkov** [comment]
@monich new PR is https://github.com/Wunderfitz/harbour-fernschreiber/pull/573

**Pull request closed without merging**